### DISCUSSION TITLE
PDFs with bad font definitions

### DISCUSSION THREAD

**Discussion by Yichen-fqyd** [pymupdf/PyMuPDF#1380] [Looking for help] [answered]
When converting this file to html, all the � for the first page, is it because the font information or the encoding of the system, is there a way to retrieve the text as it looks, for example download the font files?

Thanks!
[MLIL1370ML_A_9_2_29+(2).pdf](https://github.com/pymupdf/PyMuPDF/files/7498396/MLIL1370ML_A_9_2_29%2B.2.pdf)
s

**Comment by JorjMcKie** [comment]
did you try to output normal text via `mutool draw -o file.txt file.pdf`?
This indicates that MuPDF cannot interpret the unicode delivered by the font. There can be nothing done about it.
You **_could_** OCR the page or the problematic spans, e.g. as shown in [this](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/jupyter-notebooks/ocr-illegible.ipynb) Jupyter notebook.

**Reply by Yichen-fqyd** [reply]
Also another interesting thing about this file, when the file is converted to image using pymupdf, the result looks fine, but if convert this file through other screenshot pdf2image conversion, the image doesn't make sense. Attached two converted images from the same exact file
<img width="707" alt="Screen Shot 2021-11-08 at 2 29 40 PM" src="https://user-images.githubusercontent.com/37560330/140805456-edb7082c-df91-4ba4-b74e-d70417651c0d.png">
<img width="707" alt="Screen Shot 2021-11-08 at 2 28 32 PM" src="https://user-images.githubusercontent.com/37560330/140805478-c2db4fb2-889f-4c2b-ba60-f6267ac6dbe0.png">

**Comment by JorjMcKie** [comment, marked as answer]
Hm, this file has severe internal problem. XPDF cannot convert it to text either.
Throw it away 😎
Or OCR it.

**Comment by JorjMcKie** [comment]
You know the new OCR interface of PyMuPDF v1.19.1?
The above page could be converted - at least to some extent - to  HTML when OCRing before extracting HTML text:
```python 
tp = page.get_textpage_ocr(dpi=300, full=True)  # OCR the full page at a decent resolution
html = page.get_text("html", textpage=tp)  # use the OCRed version
out = "page.html", "w")
out.write(html)
out.close()
```
Delivers a quite decent html page.
